src/CMakeLists.txt

```diff
@@ -27,16 +27,13 @@ foreach(libtype shared static)
       )
     target_link_libraries(hiop_${libtype} PUBLIC hiop_tpl)
     set_target_properties(hiop_${libtype} PROPERTIES OUTPUT_NAME hiop)
-    install(TARGETS hiop_${libtype} DESTINATION lib)
+    install(TARGETS hiop_${libtype} DESTINATION lib EXPORT hiop-targets)
   endif()
 endforeach()
 
 # Create default library pointed to by HiOp::HiOp
 add_library(HiOp::HiOp ALIAS ${hiop_default_library_name})
```

Comment:
Will this target be exported? I think we want this to be the cmake interface for the end user so they don't concern themselves with shared/static unless they need to.

Reply:
This will be exported in `<install_dir>/share/hiop/cmake/HiOpTargets.cmake`. All TPLs and libraries that have the `EXPORT hiop-targets` are exported similarly. When I built and installed, I had the following exported for these targets:

```CMake
# Create imported target HiOp::hiop_shared
add_library(HiOp::hiop_shared SHARED IMPORTED)

set_target_properties(HiOp::hiop_shared PROPERTIES
  INTERFACE_LINK_LIBRARIES "HiOp::hiop_tpl;HiOp::Magma"
)

# Create imported target HiOp::hiop_static
add_library(HiOp::hiop_static STATIC IMPORTED)

set_target_properties(HiOp::hiop_static PROPERTIES
  INTERFACE_LINK_LIBRARIES "HiOp::hiop_tpl;HiOp::Magma"
)
```

Reply:
`HiOp::hiop_tpl` is also created as a target, and it is linked against all the libraries that `hiop_tpl` links against.

Reply:
Right, I don't really want users interacting with `_shared` or `_static` unless they have to - is `HiOp::HiOp` exported?